### DISCUSSION TITLE
Is it possible to override the an default admin REST API endpoint?

### DISCUSSION THREAD

**Discussion by initshdb** [keycloak/keycloak#20766] [Archive]
Hello,

I want to be able to override the default `/admin/serverinfo` REST API in the `AdminRoot` class to provide additional information about the server through the default `ServerInfoAdminResource`

```
@Path("/admin")
public class AdminRoot {
    ... 
   @Path("serverinfo")
    public Object getServerInfo() {
        ...

        // Want to be able to add more custom server information 
        // into the ServerInfoAdminResource may be by creating a CustomServerInfoAdminResource

        return new ServerInfoAdminResource();
    }
   ...
}
```

Is it possible to override or extend a particular default Admin Rest API?

Any input in this regards is highly appreciated.

Regards, Nitish

**Comment by twobiers** [comment]
I am also interested in an answer. In my case I would like to have an API that can search clients across all realms. As it is not very feasible to iterate over multiple realm endpoints I would like to extend the AdminRoot.

As a workaround it is possible to treat the master realm specially since as per documentation it should not be used anyway.

**Comment by danielFesenmeyer** [comment]
We have done similar things with Quarkus request/response filters: https://quarkus.io/guides/rest#request-or-response-filters

It works, but we would be interested in an easier/better way to do this.
